zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/api/ScaleUpRequestTransformer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.dynamic.config.api;
+
+import io.camunda.zeebe.dynamic.config.api.ClusterConfigurationRequestFailedException.InvalidRequest;
+import io.camunda.zeebe.dynamic.config.changes.ConfigurationChangeCoordinator.ConfigurationChangeRequest;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfiguration;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation.ScaleUpOperation.*;
+import io.camunda.zeebe.util.Either;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.stream.Stream;
+
+public class ScaleUpRequestTransformer implements ConfigurationChangeRequest {
+
+  private final int desiredPartitionCount;
+  private final SortedSet<Integer> bootstrappedPartitions;
+
+  public ScaleUpRequestTransformer(
+      final int desiredPartitionCount, final SortedSet<Integer> bootstrappedPartitions) {
+    this.desiredPartitionCount = desiredPartitionCount;
+    this.bootstrappedPartitions = bootstrappedPartitions;
+  }
+
+  @Override
+  public Either<Exception, List<ClusterConfigurationChangeOperation>> operations(
+      final ClusterConfiguration clusterConfiguration) {
+    if (desiredPartitionCount > clusterConfiguration.partitionCount()) {
```

Comment:
:wrench: I'd prefer an early return instead of the nesting inside the `else` block for readability.